### ISSUE TITLE
The PDO profile does not work during an opensim installation in standalone mode from scratch

### ISSUE THREAD

**Issue by djphil** [kcozens/OpenSimProfile#12]
I have once again tested OpenSim Profile with PDO since a new installation of OpenSim in standalone mode from scratch.
On this occasion I discovered a bug and I can reproduce it.
Here are the steps to reproduce it:

1) Create a database for a fresh opensim install
2) Setup OpenSim to run in standalone mode without Hypergrid and with OpenSimProfile
3) Add one region on Region.ini file
4) Go to your web server, create a new folder and put all php files inside
5) Edit databaseinfo.php and change values to reflet your database config
6) Go to the opensim database and add the osprofile.sql
7) Start the Simulator (during install create the new user)
8) Restart the simulator
9) Go to inworld with Singularity
10) Go to only on the first tab on your profile (2nd Life)
11) Add a new picture and click OK
12) Go back to the first tab. The picture IS NOT SAVED
Here is the bug !!!

13) Ok don't touch nothing, just try to add again a picture and clic again OK
14) Go back again to the first tab. The picture IS NOT SAVED
OK, bug confirmed !
Ok, just clic OK to go out the profile

Now new Steps (now the strange begin !!!)
15) Go to your webserver and remplace the PDO by the MySqli version
16) Now go to your profile inworld, on the first tab only, add a picture and clic OK
17) Go back to the first tab. The picture IS NOW PROPRELY SAVED, IT WORK !!!
18) Ok don't touch nothing, just go to your web server and remplace the Mysqli with the PDO
19) Now go to your profile inworld, on the first tab only and you can see the picture NOW ONLY !!!
That stange, now you can see and modify with the PDO ...

So to summarize, when you install opensim from the beginning, if you use the PDO, your profile will not work.
There must be a correct 1st record so that it works then correctly.

Here, i hope that you can reproduce also and especially you can fix it.

**Comment by djphil** [comment]
I found another way to reproduce the problem.
Simply use an avatar and empty all the profile tables of this avatar
Then try to complete profile inworld (just try adding image)
In this case the image will not be saved, the bug is here

Note that if you do exactly the same thing with the mysqli file, the image will be saved correctly. Then you can use the pdo file, it will work properly.

The problem with the pdo is only when you perform a first recording.

Do you have a track to solve this problem?

**Comment by kcozens** [comment]
commit a56e71224618c1416893c0c652c6dcbfc54b1c64
Author: Kevin Cozens <kevin@ve3syb.ca>
Date:   Tue Dec 5 12:30:10 2017 -0500

    Fixed issues with initial creation, and updating, of profile and profile picks
    Fixes issue #11 and #12.
